src/configure/configure.ts

```diff
@@ -2,41 +2,63 @@ const uuid = require('uuid/v4');
 import { AppServiceClient } from './clients/azure/appServiceClient';
 import { AzureDevOpsClient } from './clients/devOps/azureDevOpsClient';
 import { AzureDevOpsHelper } from './helper/devOps/azureDevOpsHelper';
-import { AzureTreeItem } from 'vscode-azureextensionui';
+import { AzureTreeItem, UserCancelledError } from 'vscode-azureextensionui';
 import { exit } from 'process';
 import { generateDevOpsProjectName } from './helper/commonHelper';
 import { GenericResource } from 'azure-arm-resource/lib/resource/models';
 import { GraphHelper } from './helper/graphHelper';
 import { LocalGitRepoHelper } from './helper/LocalGitRepoHelper';
-import { Messages } from './messages';
+import { Messages } from './resources/messages';
 import { QuickPickItem } from 'vscode';
 import { ServiceConnectionHelper } from './helper/devOps/serviceConnectionHelper';
 import { SourceOptions, RepositoryProvider, extensionVariables, WizardInputs, WebAppKind, PipelineTemplate, QuickPickItemWithData } from './model/models';
+import { TracePoints } from './resources/tracePoints';
 import * as path from 'path';
 import * as templateHelper from './helper/templateHelper';
 import * as utils from 'util';
 import * as vscode from 'vscode';
+import { TelemetryHelper } from './helper/telemetryHelper';
 
-export async function configurePipeline(node: any) {
+export async function configurePipeline(telemetryHelper: TelemetryHelper, node: AzureTreeItem) {
     try {
+        telemetryHelper.setTelemetry(TracePoints.CommandStartTime, Date.UTC.toString());
         if (!(await extensionVariables.azureAccountExtensionApi.waitForLogin())) {
+            // set telemetry
+            telemetryHelper.setTelemetry(TracePoints.AzureLoginRequired, 'true');
+
             let signIn = await vscode.window.showInformationMessage(Messages.azureLoginRequired, Messages.signInLabel);
-            if(signIn.toLowerCase() === Messages.signInLabel.toLowerCase()) {
+            if (signIn.toLowerCase() === Messages.signInLabel.toLowerCase()) {
                 await vscode.commands.executeCommand("azure-account.login");
             }
             else {
-                throw new Error(Messages.azureLoginRequired);
+                let error = new Error(Messages.azureLoginRequired);
+                extensionVariables.reporter.sendTelemetryEvent(
+                    TracePoints.AzureLoginFailure,
```

Comment:
I think you are mixing CI and traces here. You should log a CI telemetry to say that user did not login (use CIPoint). And if there is an error in login, then we should trace the error and use a different TracePoint.